enhancements/installer/assisted-installer-bare-metal-validations.md

```diff
@@ -0,0 +1,280 @@
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The assisted installer supports a user-provisioned infrastructure
+installation workflow for bare-metal clusters, but configures the
+installed cluster with certain capabilities that are currently only
+available using the `baremetal` platform in an installer-provisioned
+infrastructure workflow.
+
+This enhancement describes how we can modify the installer to allow
+bare-metal user-provisioned infrastructure clusters, including those
+produced with the assisted installer, to include those
+installer-provisioned infrastructure capabilities.
+
+## Motivation
+
+The [connected-assisted-installer
+enhancement](./connected-assisted-installer.md) describes how an agent
+running on several bare-metal hosts receive a command from the
+assisted installer application to begin the installation of that
+host. One of the first tasks that each host performs upon receiving
+this instruction is to download, from the assisted installer, the
+appropriate Ignition file for that host's role. These ignition files
+are generated by the assisted installer earlier in the process using
+the `openshift-install create ignition-configs` command.
+
+This installer workflow mirrors the [bare-metal user-provisioned
+infrastructure
+workflow](https://github.com/openshift/installer/blob/master/docs/user/metal/install_upi.md)
+in that the the OpenShift installer is used to generate assets for the
+cluster but the `openshift-install create cluster` command is not used
+to manage the deployment. However, the assisted installer does need to
+enable some `baremetal` installer-provisioned infrastructure platform
+features on the installed cluster. Therefore, we need to consider how
+to make those features available while using a user-provisioned
+infrastructure workflow. Currently, the assisted installer is using
+the [workaround of providing dummy
+data](https://github.com/filanov/bm-inventory/blob/54d56712b59d306125ee68d2014c98cf1e995e75/internal/installcfg/installcfg.go#L143-L190)
+for fields that are currently required by the `baremetal` platform but
+that are unused in this use-case.
+
+### Goals
+
+- Allow the use of certain installer-provisioned infrastructure
+  features on bare-metal user-provisioned infrastructure clusters
+- Avoid making any incompatible changes to the bare-metal
+  installer-provisioned infrastructure workflow
+
+### Non-Goals
+
+- Removing or relaxing any validation rules for the
+  installer-provisioned infrastructure installer.
+- Recapitulate the test plans for the assisted installer.
+- Recapitulate the [approach for making parts of the bare metal
+  infrastructure automation optional](./baremetal-provisioning-optional.md).
```

Comment:
broken link? couldn't find that doc

Reply:
[Found it](https://github.com/openshift/enhancements/blob/master/enhancements/baremetal/baremetal-provisioning-optional.md). I guess the link will work once the enhancement will be merged.